test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpath-plugin.yaml

```diff
@@ -61,15 +91,12 @@ spec:
             name: csi-data-dir
 
         - name: hostpath
-          image: k8s.gcr.io/sig-storage/hostpathplugin:v1.4.0
+          image: k8s.gcr.io/sig-storage/hostpathplugin:v1.6.2
           args:
             - "--drivername=hostpath.csi.k8s.io"
             - "--v=5"
             - "--endpoint=$(CSI_ENDPOINT)"
             - "--nodeid=$(KUBE_NODE_NAME)"
-            # The only difference to github.com/kubernetes-csi/csi-driver-host-path/deploy
-            # - we have a tests that checks node limits.
```

Comment:
I need to check that. I ran some tests locally and those worked, but not everything. The fuil test in the CI run showed that further work is needed.

Reply:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100637/pull-kubernetes-e2e-kind/1376804369592225792 had lots of other, unrelated errors. Must have been a bad host or load. I ran it again, but still no luck.

I checked some of the failed csi-hostpath tests locally and those worked for me. Some tests also passed in the Prow job, so it shouldn't be a general problem with the driver.

Testing on GCE is failing more consistently: the hostpath container doesn't come up.

Hmm, why does it merge YAML manifests as part of the job preparation?! https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/100637/pull-kubernetes-e2e-gce-csi-serial/1377185922327515136/build-log.txt

Time for a rebase...

Reply:
The merge is probably normal (merge instead of fast forward, with no other changes to those files on master).

Reply:
I found the volume limits test and moved the adding of `--maxvolumelimits` parameter to csi.go - that's where the manifests can and should be updated while loading them for tests.

Reply:
In principle the updated driver works now. What worries me a bit is the number of tests that time out - not just for storage, but also for other areas (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100637/pull-kubernetes-e2e-kind/1377330845706096640). If this is an overall problem, then the PR is ready. But if it is somehow worse with these changes, then we need to investigate further.

/retest

Reply:
I had a hunch that the new external-health-monitor sidecars might cause additional overhead, so I disabled them. This is possible because they are not actually needed for any test at the moment.

That made a striking difference:
- without them, pull-kubernetes-e2e-kind finished in 39m with just one unrelated error: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100637/pull-kubernetes-e2e-kind/1377513227537092608
- with them, the same job failed both before (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100637/pull-kubernetes-e2e-kind/1377369246706700288) and after (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100637/pull-kubernetes-e2e-kind/1377531518573875200) the one which succeeded.

That it fails consistently when the sidecars enabled at different times pretty much rules out that the state of the Prow cluster is to blame for the slowdown. Also, other PRs are getting tested (mostly) successfully.

Reply:
@xing-yang ^^^

Reply:
Or perhaps not. Now e2e-kind also failed horribly without these sidecars: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100637

Reply:
The problem seems to be elsewhere. The new hostpath driver fails to start up repeatedly while checking mount points:
```
==== start of pod log for container csi-hostpathplugin-0/hostpath ====
I0401 09:26:04.587557       1 hostpath.go:128] Driver: csi-hostpath-ephemeral-6690 
I0401 09:26:04.587617       1 hostpath.go:129] Version: v1.6.2
Failed to initialize driver: failed to get capacity info: no such file or directory
==== end of pod log for container csi-hostpathplugin-0/hostpath ====
```

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/100637/pull-kubernetes-e2e-kind/1377544760671080448/artifacts/_sig-storage_CSI_Volumes/_Driver_csi-hostpath_/_Testpattern_Generic_Ephemeral-volume_default_fs_immediate-binding_ephemeral/should_support_two_pods_which_share_the_same_volume/csi-hostpathplugin-0-hostpath.log

Unfortunately the error message is not very informative. It comes from here:

https://github.com/kubernetes-csi/csi-driver-host-path/blob/a7442639da30b79a09438e96400bc54602fc068f/pkg/hostpath/hostpath.go#L522-L540

That code is new, it was not yet in the 1.4.0 release that was used before this PR. It was added in https://github.com/kubernetes-csi/csi-driver-host-path/pull/210

@fengzixu any idea why it is failing?

Reply:
@pohly we have discussed in the CSI meeting that we don't need new e2e tests for health monitoring now because this feature only reports events on PVCs. There's no reliable way of verifying an event is reported as it can disappear.  So until we have figured out a better way, we only test this in unit tests.

Reply:
We root-caused this in https://github.com/kubernetes-csi/csi-driver-host-path/pull/210#discussion_r605592438